### PR TITLE
[DateTime] Fix code smells left from #1647

### PR DESCRIPTION
**Pull request by tobilen**
#### Fixes
* https://github.com/palantir/blueprint/pull/1647#discussion_r143568863
* https://github.com/palantir/blueprint/pull/1647#discussion_r143569040

### CODE REVIEW

File: docs/docs/site-docs.js [hunk 1]
```diff
@@ -83778,7 +83780,7 @@
 /* 665 */
 /***/ (function(module, exports) {
 
-	module.exports = [{"name":"@blueprintjs/core","version":"1.32.0"},{"name":"@blueprintjs/datetime","version":"1.23.0"},{"name":"@blueprintjs/docs","version":"1.1.1"},{"name":"@blueprintjs/labs","version":"0.12.0"},{"name":"@blueprintjs/table","version":"1.29.0"}]
+	module.exports = [{"name":"@blueprintjs/core","version":"1.32.0"},{"name":"@blueprintjs/datetime","version":"1.23.0"},{"name":"@blueprintjs/docs","version":"1.1.1"},{"name":"@blueprintjs/labs","version":"0.12.0"},{"name":"@blueprintjs/table","version":"1.29.1"}]
```
Comment (giladgray) [inline docs/docs/site-docs.js]: this file should not be changed here, only in release PRs. can you revert these changes?

File: packages/datetime/src/datePicker.tsx [hunk 2]
```diff
@@ -195,7 +195,7 @@ export class DatePicker extends AbstractComponent<IDatePickerProps, IDatePickerS
     private getDisabledDaysModifier = () => {
         const { dayPickerProps: { disabledDays } } = this.props;
 
-        return disabledDays instanceof Array ? [this.disabledDays, ...disabledDays] : [this.disabledDays, disabledDays];
+        return Array.isArray(disabledDays) ? [this.disabledDays, ...disabledDays] : [this.disabledDays, disabledDays];
```
Comment (giladgray) [inline packages/datetime/src/datePicker.tsx]: [`Array.prototype.concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat?v=b) actually handles both cases rather neatly!

`return this.disabledDays.concat(disabledDays);`

File: packages/table/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@blueprintjs/table",
-  "version": "1.29.0",
+  "version": "1.29.1",
```
Comment (giladgray) [inline packages/table/package.json]: these two package.json changes also don't belong here... perhaps you branched off something other than the latest `master`?

**Comment by giladgray** [comment]
@tobilen thanks for addressing those dangling comments! very much appreciated 🍰  👍

File: packages/datetime/src/datePicker.tsx (hunk 2 above)
Comment (tobilen) [reply packages/datetime/src/datePicker.tsx]: thats something @cmslewis suggested already. typescript compiler complains when doing this.

**Comment by tobilen** [comment]
<h3>Roll back accidental table version increment</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/tobilen/blueprint/15/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/tobilen/blueprint/15/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/tobilen/blueprint/15/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/tobilen/blueprint/15/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by tobilen** [comment]
<h3>Roll back accidental table version increment</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/tobilen/blueprint/16/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/tobilen/blueprint/16/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/tobilen/blueprint/16/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/tobilen/blueprint/16/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by tobilen** [comment]
<h3>Revert "Prepare Release table-1.29.1 (#1742)"</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/tobilen/blueprint/17/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/tobilen/blueprint/17/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/tobilen/blueprint/17/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/tobilen/blueprint/17/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by cmslewis** [comment]
@giladgray this is ready for a second pass

File: packages/datetime/src/datePicker.tsx (hunk 2 above)
Comment (giladgray) [reply packages/datetime/src/datePicker.tsx]: oh sure, cuz `this.disabledDays` isn't necessarily an array. ok this is fine.

**Comment by giladgray** [comment]
thank you @tobilen!
